src/common/config.h

```diff
@@ -108,24 +51,21 @@ class Config {
       return dflt;
   }
 
+  const YAML::Node& get() const;
+  YAML::Node& get();
+
   template <typename T>
   void set(const std::string& key, const T& value) {
     config_[key] = value;
   }
 
-  const YAML::Node& get() const;
-  YAML::Node& get();
-
   YAML::Node getModelParameters();
   void loadModelParameters(const std::string& name);
   void loadModelParameters(const void* ptr);
 
-  const std::vector<DeviceId>& getDevices() { return devices_; }
+  const std::vector<DeviceId>& getDevices();
 
-  void save(const std::string& name) {
-    OutputFileStream out(name);
-    (std::ostream&)out << *this;
-  }
+  void save(const std::string& name);
```

Comment:
Is this save() function now more complex than before? If not, it may be appropriate to keep the original.